### PR TITLE
glaze: add version 1.9.8.1

### PR DESCRIPTION
**Pull request by toge**
Specify library name and version:  **glaze/1.9.8.1**

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by toge** [comment]
strange.

glaze/1.9.7 works fine on Visual Studio 19.37.32824.0 on my environment.


<details>
<summary>conan create log</summary>

```txt
======== Exporting recipe to the cache ========
glaze/1.9.7: Exporting package recipe: C:\Users\toge\src\conan-center-index\recipes\glaze\all\conanfile.py
glaze/1.9.7: exports: File 'conandata.yml' found. Exporting it...
glaze/1.9.7: Copied 1 '.yml' file: conandata.yml
glaze/1.9.7: Copied 1 '.py' file: conanfile.py
glaze/1.9.7: Exported to cache folder: C:\Users\toge\.conan2\p\glaze6765c5d9204eb\e
glaze/1.9.7: Exported: glaze/1.9.7#8417a6f9831aaabbbf1bffb33c272bb9 (2023-12-18 18:00:32 UTC)

======== Input profiles ========
Profile host:
[settings]
arch=x86_64
build_type=Release
compiler=msvc
compiler.cppstd=20
compiler.runtime=dynamic
compiler.runtime_type=Release
compiler.version=193
os=Windows

Profile build:
[settings]
arch=x86_64
build_type=Release
compiler=msvc
compiler.cppstd=20
compiler.runtime=dynamic
compiler.runtime_type=Release
compiler.version=193
os=Windows


======== Computing dependency graph ========
Graph root
    cli
Requirements
    glaze/1.9.7#8417a6f9831aaabbbf1bffb33c272bb9 - Cache

======== Computing necessary packages ========
glaze/1.9.7: Compatible package ID da39a3ee5e6b4b0d3255bfef95601890afd80709 equal to the default package ID
Requirements
    glaze/1.9.7#8417a6f9831aaabbbf1bffb33c272bb9:da39a3ee5e6b4b0d3255bfef95601890afd80709 - Build

======== Installing packages ========
glaze/1.9.7: Calling source() in C:\Users\toge\.conan2\p\glaze6765c5d9204eb\s\src

-------- Installing package glaze/1.9.7 (1 of 1) --------
glaze/1.9.7: Building from source
glaze/1.9.7: Package glaze/1.9.7:da39a3ee5e6b4b0d3255bfef95601890afd80709
glaze/1.9.7: Building your package in C:\Users\toge\.conan2\p\b\glazed62b7b358165d\b
glaze/1.9.7: Generating aggregated env files
glaze/1.9.7: Generated aggregated env files: ['conanbuild.bat', 'conanrun.bat']
glaze/1.9.7: Package 'da39a3ee5e6b4b0d3255bfef95601890afd80709' built
glaze/1.9.7: Build folder C:\Users\toge\.conan2\p\b\glazed62b7b358165d\b\build-release
glaze/1.9.7: Generating the package
glaze/1.9.7: Packaging in folder C:\Users\toge\.conan2\p\b\glazed62b7b358165d\p
glaze/1.9.7: Calling package()
glaze/1.9.7: package(): Packaged 98 '.hpp' files
glaze/1.9.7: package(): Packaged 1 file: LICENSE
glaze/1.9.7: Created package revision 0e0eb283a28b8ad21769859ca17a5969
glaze/1.9.7: Package 'da39a3ee5e6b4b0d3255bfef95601890afd80709' created
glaze/1.9.7: Full package reference: glaze/1.9.7#8417a6f9831aaabbbf1bffb33c272bb9:da39a3ee5e6b4b0d3255bfef95601890afd80709#0e0eb283a28b8ad21769859ca17a5969
glaze/1.9.7: Package folder C:\Users\toge\.conan2\p\b\glazed62b7b358165d\p

======== Launching test_package ========

======== Computing dependency graph ========
Graph root
    glaze/1.9.7 (test package): C:\Users\toge\src\conan-center-index\recipes\glaze\all\test_package\conanfile.py
Requirements
    glaze/1.9.7#8417a6f9831aaabbbf1bffb33c272bb9 - Cache

======== Computing necessary packages ========
Requirements
    glaze/1.9.7#8417a6f9831aaabbbf1bffb33c272bb9:da39a3ee5e6b4b0d3255bfef95601890afd80709#0e0eb283a28b8ad21769859ca17a5969 - Cache

======== Installing packages ========
glaze/1.9.7: Already installed! (1 of 1)

======== Testing the package ========
Removing previously existing 'test_package' build folder: C:\Users\toge\src\conan-center-index\recipes\glaze\all\test_package\build\msvc-193-x86_64-20-release
glaze/1.9.7 (test package): Test package build: build\msvc-193-x86_64-20-release
glaze/1.9.7 (test package): Test package build folder: C:\Users\toge\src\conan-center-index\recipes\glaze\all\test_package\build\msvc-193-x86_64-20-release
glaze/1.9.7 (test package): Writing generators to C:\Users\toge\src\conan-center-index\recipes\glaze\all\test_package\build\msvc-193-x86_64-20-release\generators
glaze/1.9.7 (test package): Generator 'CMakeDeps' calling 'generate()'
glaze/1.9.7 (test package): Generator 'CMakeToolchain' calling 'generate()'
glaze/1.9.7 (test package): CMakeToolchain generated: conan_toolchain.cmake
glaze/1.9.7 (test package): CMakeToolchain generated: CMakePresets.json
glaze/1.9.7 (test package): CMakeToolchain generated: ..\..\..\CMakeUserPresets.json
glaze/1.9.7 (test package): Generator 'VirtualRunEnv' calling 'generate()'
glaze/1.9.7 (test package): Generating aggregated env files
glaze/1.9.7 (test package): Generated aggregated env files: ['conanrun.bat', 'conanbuild.bat']

======== Testing the package: Building ========
glaze/1.9.7 (test package): Calling build()
glaze/1.9.7 (test package): Running CMake.configure()
glaze/1.9.7 (test package): RUN: cmake -G "Visual Studio 17 2022" -DCMAKE_TOOLCHAIN_FILE="C:/Users/toge/src/conan-center-index/recipes/glaze/all/test_package/build/msvc-193-x86_64-20-release/generators/conan_toolchain.cmake" -DCMAKE_INSTALL_PREFIX="C:/Users/toge/src/conan-center-index/recipes/glaze/all/test_package" -DCMAKE_POLICY_DEFAULT_CMP0091="NEW" "C:\Users\toge\src\conan-center-index\recipes\glaze\all\test_package"
-- Using Conan toolchain: C:/Users/toge/src/conan-center-index/recipes/glaze/all/test_package/build/msvc-193-x86_64-20-release/generators/conan_toolchain.cmake
-- Conan toolchain: C++ Standard 20 with extensions OFF
-- Selecting Windows SDK version 10.0.20348.0 to target Windows 10.0.19045.
-- The CXX compiler identification is MSVC 19.37.32824.0
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.37.32822/bin/Hostx64/x64/cl.exe - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Conan: Target declared 'glaze::glaze'
-- Configuring done (2.3s)
-- Generating done (0.0s)
-- Build files have been written to: C:/Users/toge/src/conan-center-index/recipes/glaze/all/test_package/build/msvc-193-x86_64-20-release

glaze/1.9.7 (test package): Running CMake.build()
glaze/1.9.7 (test package): RUN: cmake --build "C:\Users\toge\src\conan-center-index\recipes\glaze\all\test_package\build\msvc-193-x86_64-20-release" --config Release
MSBuild のバージョン 17.7.2+d6990bcfa (.NET Framework)

  1>Checking Build System
  Building Custom Rule C:/Users/toge/src/conan-center-index/recipes/glaze/all/test_package/CMakeLists.txt
  test_package.cpp
C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.37.32822\include\expected(12): warning STL4
038: The contents of <expected> are available only with C++23 or later. [C:\Users\toge\src\conan-center-index\recipe
s\glaze\all\test_package\build\msvc-193-x86_64-20-release\test_package.vcxproj]
  test_package.vcxproj -> C:\Users\toge\src\conan-center-index\recipes\glaze\all\test_package\build\msvc-193-x86_64-
  20-release\Release\test_package.exe
  Building Custom Rule C:/Users/toge/src/conan-center-index/recipes/glaze/all/test_package/CMakeLists.txt


======== Testing the package: Executing test ========
glaze/1.9.7 (test package): Running test()
glaze/1.9.7 (test package): RUN: Release\test_package
```
</details>

File: recipes/glaze/all/conanfile.py [hunk 1]
```diff
@@ -51,6 +51,13 @@ def validate(self):
     def source(self):
         get(self, **self.conan_data["sources"][self.version], strip_root=True)
 
+    def build(self):
+        if Version(self.version) >= "1.9.7":
+            replace_in_file(self, os.path.join(self.source_folder, "include", "glaze", "csv", "write.hpp"),
```
Comment (uilianries) [inline recipes/glaze/all/conanfile.py:56]: Hello @toge! It seems like this patch is related to the msvc failure: https://github.com/conan-io/conan-center-index/pull/21748#issuecomment-1859597357 

It looks weird that error only happening in CCI, but taking a look on https://github.com/stephenberry/glaze/issues, I see more related issues not to std:;decay, but other C++ features. 

I would request sending an issue to discuss with the upstream first, or sending a PR with that fix. 

Regards.

Comment (toge) [reply recipes/glaze/all/conanfile.py:56]: I agree with you.
It seems to be a msvc bug.
It is almost same as following. 
https://developercommunity.visualstudio.com/t/164-regression-MSVC-1424-gets-intern/848410?q=std%3A%3Adecay_t.

**Review by ericLemanissier — APPROVED** [review]

File: recipes/glaze/all/conanfile.py (hunk 1 above)
Comment (toge) [reply recipes/glaze/all/conanfile.py]: @uilianries 
Fixed.

Comment (uilianries) [reply recipes/glaze/all/conanfile.py]: @toge, thank you for pointing it!

**Review by uilianries — APPROVED** [review]
LGTM

**Pull request merged**